### PR TITLE
PR: Update contributing, release and maintenance instructions for backporting

### PR DESCRIPTION
**Pull request by ccordoba12**
## Description of Changes

This not only simplifies our workflow by avoiding the need to manually forward-port PRs from the stable branch to master. It'll also help us to maintain a more stable branch, by only backporting what we think is really necessary. 

### Issue(s) Resolved

Fixes #

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct: @ccordoba12

### CODE REVIEW

**Comment by ccordoba12** [comment]
@dalthviz, this is ready for review.

File: RELEASE.md [hunk 1]
```diff
@@ -203,16 +229,12 @@ To release a new version of Spyder you need to follow these steps:
 
 * `git tag -a vX.X.X -m "Release X.X.X"`
 
-* Update version in `__init__.py` (add 'dev0' and increment minor)
+* Update version in `__init__.py` (add 'a1', 'dev0' and increment patch version)
```
Comment (dalthviz) [inline RELEASE.md:232]: Will `a1` be always corresponding or an instruction to update that needs to be added to?

**Comment by dalthviz** [comment]
Give a quick check and makes sense to me :+1: Left only a single comment related with the management of the alpha part over the version update. Also, thinking about docs updates. Could this PR be a good place to update the README badges that launch binder instances?

image

Not sure if there are other places where 5.x is still being mentioned 🤔

File: RELEASE.md (hunk 1 above)
Comment (ccordoba12) [reply RELEASE.md:232]: Nop, that's the case only for stable versions. I'll update the instruction to cover pre-releases as well.

**Comment by ccordoba12** [comment]
> Could this PR be a good place to update the README badges that launch binder instances?

I'll leave that for another PR to include a critical fix to avoid hard freezes on Binder.

**Review by dalthviz — APPROVED** [review]
Thanks @ccordoba12 ! LGTM :+1: Following the updated steps, after this gets merged, this should be backported to 6.x, right?

Also, is there any other change you want to add here or should I proceed with the merge?

**Comment by ccordoba12** [comment]
> Following the updated steps, after this gets merged, this should be backported to 6.x, right?

Correct. If a PR is marked for 6.0.x, 6.1.x, etc, it needs to be backported.

> Also, is there any other change you want to add here or should I proceed with the merge?

Nop, this is ready.

By the way, I'm using regular merges in master to preserve the commit history, but squash merges in 6.x because history is not so critical there. And I'm planning to add another bot to automatically do that in 6.x.

**Comment by dalthviz** [comment]
@meeseeksdev please backport to 6.x

**Comment by dalthviz** [comment]
> By the way, I'm using regular merges in master to preserve the commit history, but squash merges in 6.x because history is not so critical there. And I'm planning to add another bot to automatically do that in 6.x.

Is there a case where you will need to merge something directly to 6.x without passing from master and the backporting procedure (which seems like creates a PR with a single commit)?

**Comment by ccordoba12** [comment]
> Is there a case where you will need to merge something directly to 6.x without passing from master and the backporting procedure (which seems like creates a PR with a single commit)?

There should be very few things that only apply to `6.x`. For instance, I made PR #22468 to fix its dev version because that couldn't be backported (PR #22465 was the equivalent for master).

**Pull request merged**